app.rb

```diff
@@ -0,0 +1,138 @@
+require_relative 'book'
+require_relative 'classroom'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'Currently, there are no books!'
+    else
+      @books.each_with_index do |book, i|
+        puts "#{i}) Title: #{book.title}, Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'Currently, there are no people in this library / school!'
+    else
+      @people.each_with_index do |person, i|
+        puts "#{i}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def add_student
+    puts 'Age: '
+    age = gets.chomp.to_i
+
+    puts 'Name: '
+    name = gets.chomp.to_s
+
+    puts 'Has parent permission? [Y / N]: '
+    parent_permission = gets.chomp.to_s
+
+    puts 'Class:'
+    classroom = gets.chomp.to_s
+
+    if parent_permission =~ /^[Yy]/
+      @people.push(Student.new(classroom, age, name, parent_permission: true))
+    elsif parent_permission =~ /^[Nn]/
+      @people.push(Student.new(classroom, age, name, parent_permission: false))
+    else
+      puts "Invalid choice. Please enter a valid option. (#{parent_permission})"
+    end
+  end
+
+  def add_teacher
+    puts 'Age: '
+    age = gets.chomp.to_i
+
+    puts 'Name: '
+    name = gets.chomp.to_s
+
+    puts 'Specialization: '
+    specialization = gets.chomp.to_s
+
+    @people.push(Teacher.new(specialization, age, name))
+  end
+
+  def add_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [input the number]: '
+    person_type = gets.chomp.to_i
+
+    case person_type
+    when 1
+      add_student
+    when 2
+      add_teacher
+    else
+      puts 'Invalid choice. Please enter a valid option.'
+    end
+
+    puts 'Person created successfully'
+  end
+
+  def add_book
+    puts 'Title:'
+    title = gets.chomp.to_s
+
+    puts 'Author:'
+    author = gets.chomp.to_s
+
+    @books.push(Book.new(title, author))
+  end
+
+  def add_rental
```

Comment:
- [ ] Kindly fix this error in the `Create a rental` option

![image](https://github.com/Arch-Noize/oop-library/assets/24703171/7485e475-3049-4e4d-9ecc-f04cbfe4f150)

When I try to create a new rental I get this error.